src/utils/sort.ts

```diff
@@ -0,0 +1,408 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TextEdit} from 'vscode-languageserver-textdocument';
+import { createScanner, SyntaxKind, JSONScanner } from 'jsonc-parser';
+import { TextDocument, FormattingOptions } from '../jsonLanguageTypes';
+import { format } from './format';
+import { PropertyTree, Container} from './propertyTree';
+
+export function sort(documentToSort: TextDocument, formattingOptions: FormattingOptions): string {
+    const options = { 
+        tabSize: formattingOptions.tabSize ? formattingOptions.tabSize : 4,
+        insertFinalNewline: formattingOptions.insertFinalNewline ? formattingOptions.insertFinalNewline : false,
+        insertSpaces: formattingOptions.insertFinalNewline ? formattingOptions.insertFinalNewline : true, 
+        keepLines: false, // keepLines must be false so that the properties are on separate lines for the sorting
+        eol: '\n'
+    };
+    let formattedJSONString: string = TextDocument.applyEdits(documentToSort, format(documentToSort, options, undefined));
+    const arrayOfLines: string[] = formattedJSONString.split('\n');
+    const jsonPropertyTree : PropertyTree = findJSONPropertyTree(formattedJSONString);
+    const sortedArrayOfLines = sortLinesOfArray(arrayOfLines, jsonPropertyTree);
+    const sortedDocument: TextDocument = TextDocument.create('test://test.json', 'json', 0, sortedArrayOfLines.join('\n'));
+    const edits: TextEdit[] = format(sortedDocument, options, undefined);
+    return TextDocument.applyEdits(sortedDocument, edits);
+}
+
+function findJSONPropertyTree(formattedString : string) {
+
+    const arrayOfLines: string[] = formattedString.split('\n');
```

Comment:
It's better to avoid breaking up the file by lines and creating an array. Line delimiters can be `\n`, `\r` and '\r\n'.
Instead use the TextDocument to handle that.
The text document allows to access the content by line.